lib/Doctrine/DataDict/Mysql.php

```diff
@@ -146,23 +146,15 @@ public function getNativeDeclaration($field)
 
                 return $length ? 'CHAR(' . $length . ')' : 'CHAR(255)';
             case 'enum':
```

Comment:
Code for ENUM and SET was identical here, except for the constant check, so just moving into a single block with an added if condition on that constant.